File: p2/p2.py
import numpy as np


def fib(n):
    if n==1:
        return 1
    elif n == 2: 
        return 2
    else:
        return fib(n-1) + fib(n-2)

def fastFib(n):
    old, new = 1, 1
    if n == 1:
        return 1
    for i in range(n-1):
        old, new = new, old + new
    return new

def evenList(numL):
    numL = np.array(numL)
    return 1 - numL%2

def ans():
    fibL = []
    n = 1
    while fib(n) < 4*10**6:
        fibL += [fib(n)]
        n += 1
    fibL = np.array(fibL)
    ev_fibL = evenList(fibL)
    answer = np.dot(fibL, ev_fibL)
    return answer 

def fast_ans():
    fibL = []
    n = 1
    while fastFib(n) < 4*10**6:
        fibL += [fastFib(n)]
        n += 1
    fibL = np.array(fibL)
    ev_fibL = evenList(fibL)
    answer = np.dot(fibL, ev_fibL)
    return answer 

File: p2/p2.jl
using LinearAlgebra

function fib(n)
	if n == 1
		return 1
	elseif n == 2
		return 2
	else
		return fib(n-1) + fib(n-2)
	end	
end


function fastFib(n)
	if n==1
		return 1
	end
	old, new = 1,1 
	for i = 1:n-1
		old, new = new, old + new
	end	
	return new
end


function p2ans()
	n = 1
	fibL = []
	while fib(n) < 4*10^6
		push!(fibL, fib(n))
		n+=1	
	end	
	evList = 1 .- fibL.%2
	return dot(vec(evList), vec(fibL))
end


function fastans()
	n = 1
	fibL = []
	while fastFib(n) < 4*10^6
		push!(fibL, fastFib(n))
		n+=1
	end
	evList = 1 .- fibL.%2
	return dot(vec(evList), vec(fibL))
end
